utilities/include/vec3.hpp

```diff
@@ -1,212 +1,59 @@
 #ifndef VEC3_HPP
 #define VEC3_HPP
 
-
 #include <iostream>
-#include <cmath> //constexpr is compiler specific
-class vec3{
-    float x,y,z;
-    public:
-        constexpr vec3() : x {0}, y {0}, z {0} {}
-
-        //Cartesian coordinate Constructor
-        constexpr vec3(float X,float Y,float Z) : x {X}, y {Y}, z {Z} {}
-
-        //norm function
-        #ifndef __GNUC__
-
-        float norm() const {
-            return sqrt(x * x + y * y + z * z);
-        }
-
-        #endif
-
-        #ifdef __GNUC__
-        
-        constexpr float norm() const {
-            return sqrt(x * x + y * y + z * z);
-        }
-
-        #endif
-
-
-
-        //unit vector function
-        constexpr vec3 unit() const{
-            return *this/this->norm();
-        }
-
-        //getters
-        constexpr float X() const{
-            return x;
-        }
+#include <cmath>
+#include <type_traits>
+
+#include "vec4.hpp"
+#include "config.hpp"
+
+class alignas(ALIGN_WIDTH) Vec3 : public Vec4{
+    private:
+                
+        /*
+        structs as_index,indices are used to generate appropriate xsimd::batch_constant 
+        to be used for xsimd::swizzle function used inside cross product. 
+        */
+        template <class T>
+        struct as_index
+        {
+            using type = xsimd::as_unsigned_integer_t<T>;
+        };
+        struct indices
+        {
+            static constexpr unsigned get(unsigned i, unsigned n)
```

Comment:
Write `unsigned int` instead of just `unsigned`. The fact that when you don't specify an integer type along with the `unsigned` keyword, the compiler deduces it to be an `int`, is a bit of trivia and reading this code (in its current form) will trip many people up.